gooddata-sdk/gooddata_sdk/catalog/workspace/entity_model/filter_view.py

```diff
@@ -0,0 +1,147 @@
+# (C) 2024 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any, Optional
+
+import attr
+from gooddata_api_client.model.json_api_filter_view_in import JsonApiFilterViewIn
+from gooddata_api_client.model.json_api_filter_view_in_attributes import JsonApiFilterViewInAttributes
+from gooddata_api_client.model.json_api_filter_view_in_document import JsonApiFilterViewInDocument
+from gooddata_api_client.model.json_api_filter_view_in_relationships import JsonApiFilterViewInRelationships
+
+from gooddata_sdk.catalog.base import Base
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class CatalogFilterViewDocument(Base):
+    data: CatalogFilterView
+
+    @staticmethod
+    def client_class() -> type[JsonApiFilterViewInDocument]:
+        return JsonApiFilterViewInDocument
+
+    def to_api(self) -> JsonApiFilterViewInDocument:
+        return JsonApiFilterViewInDocument(data=self.data.to_api())
+
+
+def _data_entity(value: Any) -> dict[str, Any]:
+    return {"data": value}
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class CatalogFilterView(Base):
+    id: Optional[str] = None
+    attributes: CatalogFilterViewAttributes
+    relationships: Optional[CatalogFilterViewRelationships] = None
+
+    @staticmethod
+    def client_class() -> type[JsonApiFilterViewIn]:
+        return JsonApiFilterViewIn
+
+    @classmethod
+    def init(
+        cls,
+        filter_view_id: str,
+        content: dict[str, Any],
+        title: str,
+        are_relations_valid: Optional[bool] = None,
+        description: Optional[str] = None,
+        is_default: Optional[bool] = None,
+        tags: Optional[list[str]] = None,
+        user_id: Optional[str] = None,
+        analytical_dashboard_id: Optional[str] = None,
+    ) -> CatalogFilterView:
+        attributes = CatalogFilterViewAttributes(
+            content=content,
+            title=title,
+            are_relations_valid=are_relations_valid,
+            description=description,
+            is_default=is_default,
+            tags=tags,
+        )
+        relationships = CatalogFilterViewRelationships.create_user_analytical_dashboard_relationship(
+            user_id=user_id, analytical_dashboard_id=analytical_dashboard_id
+        )
+        return cls(id=filter_view_id, attributes=attributes, relationships=relationships)
+
+    def to_api(self) -> JsonApiFilterViewIn:
+        attributes = self.attributes.to_api()
+        relationships = self.relationships.to_api() if self.relationships is not None else None
+        return JsonApiFilterViewIn(id=self.id, attributes=attributes, relationships=relationships)
+
+    @property
+    def user_id(self) -> str | None:
+        if self.relationships and self.relationships.user:
+            return self.relationships.user["data"].id
+        return None
+
+    @property
+    def analytical_dashboard_id(self) -> str | None:
```

Comment:
`str | None` is supported since 3.10. See [PEP-604](https://peps.python.org/pep-0604/). Strangely, mypy did not catch this. Please, use `Union[str, None] `instead.